src/hooks/useWalletsHiddenBalances.ts

```diff
@@ -0,0 +1,106 @@
+import { AllRainbowWallets } from '@/model/wallet';
+import { useMemo, useState, useEffect, useCallback } from 'react';
+import { Address } from 'viem';
+import useAccountSettings from './useAccountSettings';
+import { useConnectedToHardhatStore } from '@/state/connectedToHardhat';
+import { NativeCurrencyKey } from '@/entities/nativeCurrencyTypes';
+import { userAssetsStore } from '@/state/assets/userAssets';
+import { queryClient } from '@/react-query';
+import { userAssetsQueryKey, UserAssetsResult } from '@/resources/assets/UserAssetsQuery';
+import { convertAmountAndPriceToNativeDisplay, add } from '@/helpers/utilities';
+import { isEqual } from 'lodash';
+
+export type WalletBalanceResult = {
+  hiddenBalances: Record<Address, string>;
+};
+
+const getHiddenAssetBalance = ({
+  address,
+  nativeCurrency,
+  connectedToHardhat,
+}: {
+  address: Address;
+  nativeCurrency: NativeCurrencyKey;
+  connectedToHardhat: boolean;
+}) => {
+  const { hiddenAssets } = userAssetsStore.getState(address);
+  const assetData = queryClient.getQueryData<UserAssetsResult>(
+    userAssetsQueryKey({ address, currency: nativeCurrency, connectedToHardhat })
+  );
+
+  return Array.from(hiddenAssets).reduce((acc, uniqueId) => {
+    const asset = assetData?.[uniqueId];
+    let sum = acc;
+    if (asset) {
+      const priceUnit = asset.price?.value ?? 0;
+      const nativeDisplay = convertAmountAndPriceToNativeDisplay(asset?.balance?.amount ?? 0, priceUnit, nativeCurrency);
+      sum = add(sum, nativeDisplay.amount);
+    }
+    return sum;
+  }, '0');
+};
+
+const useWalletsHiddenBalances = (wallets: AllRainbowWallets): WalletBalanceResult => {
+  const { nativeCurrency } = useAccountSettings();
+  const connectedToHardhat = useConnectedToHardhatStore(state => state.connectedToHardhat);
+  const [hiddenBalances, setHiddenBalances] = useState<Record<Address, string>>({});
+
+  const allAddresses = useMemo(
+    () => Object.values(wallets).flatMap(wallet => (wallet.addresses || []).map(account => account.address as Address)),
+    [wallets]
+  );
+
+  const calculateHiddenBalanceForAddress = useCallback(
+    (address: Address) => {
+      const lowerCaseAddress = address.toLowerCase() as Address;
+      const hiddenAssetBalance = getHiddenAssetBalance({ address, nativeCurrency, connectedToHardhat });
+
+      console.log('calculateHiddenBalanceForAddress', address, hiddenAssetBalance);
+      setHiddenBalances(prev => ({
+        ...prev,
+        [lowerCaseAddress]: hiddenAssetBalance,
+      }));
+    },
+    [nativeCurrency, connectedToHardhat]
+  );
+
+  useEffect(() => {
+    allAddresses.forEach(calculateHiddenBalanceForAddress);
+  }, [allAddresses, calculateHiddenBalanceForAddress]);
+
+  // TODO: This is not working as intended. .subscribe() does not trigger when the value changes.
```

Comment:
@benisgold could use your help here if you have any ideas why this might not be working